libcst/metadata/scope_provider.py

```diff
@@ -584,10 +596,17 @@ def __init__(self, provider: "ScopeProvider") -> None:
 
     @contextmanager
     def _new_scope(
-        self, kind: Type[LocalScope], name: Optional[str] = None
+        self,
+        kind: Type[LocalScope],
+        name: Optional[str] = None,
+        node: Optional[cst.CSTNode] = None,
     ) -> Iterator[None]:
         parent_scope = self.scope
-        self.scope = kind(parent_scope, name)
+        if node and kind in (FunctionScope, ClassScope):
+            # pyre-ignore: pyre don't know FunctionScope and ClassScope take node arg.
```

Comment:
It seems this pyre error is alluding to the fact that these now have different constructors. I think its better to have the node arg across all constructors and do nothing with it in those that don't care.

Reply:
Add the node to all LocalScope.